projects/super_res/super_res.py

```diff
@@ -0,0 +1,203 @@
+
+
+def upscale_nearest(ds, upscale_factor, dim):
+    ds_upscaled = xr.apply_ufunc(
+        repeat_arr,
+        ds,
+        input_core_dims=[[dim]],  # apply function along this dimension
+        output_core_dims=[[dim]],  # the output will also have this dimension
+        kwargs={
+            "upscale_factor": upscale_factor
+        },  # pass in other arguments to the function
+        dask="parallelized",  # needed for dask arrays
+        output_sizes={
+            dim: ds.sizes[dim] * upscale_factor
+        },  # the size of the output dimension
+        exclude_dims=set((dim,)),  # allow changing the dimension size
+    )
+    return ds_upscaled
+
+
+def plot_pattern_bias(output: xr.Dataset, target: xr.Dataset, varname: str):
+    """Plot the pattern bias between two datasets
+
+    Args:
+        output (xr.Dataset): output dataset
+        target (xr.Dataset): target dataset
+    """
+    # compute the pattern bias
+    out_mean: xr.Dataset = output.mean("time")
+    target_mean: xr.Dataset = target.mean("time")
+    vmin = min(
+        # must convert min() output dask array to float explicitly,
+        # or we get an exception when it's used in plot_cube
+        out_mean[varname].min().values.item(),
+        target_mean[varname].min().values.item(),
+    )
+    vmax = max(
+        out_mean[varname].max().values.item(), target_mean[varname].max().values.item(),
+    )
+    bias: xr.Dataset = out_mean - target_mean
+    fig, ax = plt.subplots(
+        3, 1, figsize=(6, 9), subplot_kw={"projection": ccrs.Robinson()}
+    )
+    # plot the pattern bias
+    fv3viz.plot_cube(
+        out_mean.merge(GRID, compat="override"),
+        var_name=varname,
+        ax=ax[0],
+        vmin=vmin,
+        vmax=vmax,
+    )
+    ax[0].set_title("Time-mean Output")
+    fv3viz.plot_cube(
+        target_mean.merge(GRID, compat="override"),
+        var_name=varname,
+        ax=ax[1],
+        vmin=vmin,
+        vmax=vmax,
+    )
+    ax[1].set_title("Time-mean Target")
+    fv3viz.plot_cube(
+        # must merge grid data into the dataset
+        # after performing bias calculation on dataset
+        bias.merge(GRID, compat="override"),
+        var_name=varname,
+        cmap="RdBu_r",
+        ax=ax[2],
+    )
+    # TODO: add area-weighting to bias calculation
+    pattern_rmse = np.sqrt((bias[varname] ** 2).mean().values.item())
+    pattern_mean = bias[varname].mean().values.item()
+    ax[2].set_title(
+        "Pattern Bias, mean: {:.2f}, RMSE: {:.2f}".format(pattern_mean, pattern_rmse)
+    )
+    plt.tight_layout()
+    fig.savefig(f"plots/{varname}-pattern_bias.png", dpi=DPI)
+
+
+def plot_first_snapshot(output: xr.Dataset, target: xr.Dataset, varname: str):
+    # must specify projection for subplots if calling plot_cube with an ax argument
+    fig, ax = plt.subplots(
+        3, 1, figsize=(6, 9), subplot_kw={"projection": ccrs.Robinson()}
+    )
+    output_snapshot = output.isel(time=0, drop=True)
```

Comment:
Maybe I'm missing something but is there something special about the first snapshot, e.g. `time=0`? Maybe it would make sense to make this an argument to this function?

Reply:
It can. I just included this as a quick example to look at what the model does in one timestep.